integration/mesos/src/test/java/alluxio/mesos/AlluxioFrameworkIntegrationTest.java

```diff
@@ -0,0 +1,203 @@
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.mesos;
+
+import static org.junit.Assert.assertEquals;
+
+import alluxio.AlluxioURI;
+import alluxio.CommonTestUtils;
+import alluxio.Configuration;
+import alluxio.Constants;
+import alluxio.PropertyKey;
+import alluxio.client.ClientContext;
+import alluxio.client.block.BlockMasterClient;
+import alluxio.client.block.RetryHandlingBlockMasterClient;
+import alluxio.client.file.FileInStream;
+import alluxio.client.file.FileOutStream;
+import alluxio.client.file.FileSystem;
+import alluxio.exception.ConnectionFailedException;
+import alluxio.util.CommonUtils;
+import alluxio.util.io.PathUtils;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.google.common.base.Function;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.io.IOUtils;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * Integration tests for {@link AlluxioFramework}. These tests assume that a Mesos cluster is
+ * running locally.
+ */
+public class AlluxioFrameworkIntegrationTest {
+  private static final String JDK_URL =
+      "https://s3-us-west-2.amazonaws.com/alluxio-mesos/jdk-7u79-macosx-x64.tar.gz";
+  private static final String JDK_PATH = "jdk1.7.0_79.jdk/Contents/Home";
+
+  @Parameter(names = {"-m", "--mesos"}, required = true,
+      description = "Address for locally-running Mesos, e.g. localhost:5050")
+  private String mMesosAddress;
+
+  @Parameter(names = {"-a", "--alluxio"},
+      description = "URL of an Alluxio tarball to test. Otherwise only test local Alluxio")
+  private String mAlluxioUrl;
+
+  @Parameter(names = {"-h", "--help"}, help = true)
+  private boolean help = false;
+
+  public AlluxioFrameworkIntegrationTest() {}
+
+  private void run() throws Exception {
+    checkMesosRunning();
+    stopAlluxio();
+    stopAlluxioFramework();
+    runTests();
+    System.out.println("All tests passed!");
+  }
+
+  private void runTests() throws Exception {
+    System.out.println("Testing deployment with preinstalled alluxio and jdk");
+    testMesosDeploy(ImmutableMap.of(
+        PropertyKey.INTEGRATION_MESOS_JDK_URL, Constants.MESOS_LOCAL_INSTALL,
+        PropertyKey.INTEGRATION_MESOS_ALLUXIO_JAR_URL, Constants.MESOS_LOCAL_INSTALL));
+    System.out.println("Testing deployment with downloaded jdk");
+    testMesosDeploy(ImmutableMap.of(
+        PropertyKey.INTEGRATION_MESOS_JDK_URL, JDK_URL,
+        PropertyKey.INTEGRATION_MESOS_ALLUXIO_JAR_URL, Constants.MESOS_LOCAL_INSTALL,
+        PropertyKey.INTEGRATION_MESOS_JDK_PATH, JDK_PATH));
+    if (mAlluxioUrl != null) {
```

Comment:
I think the value is marginal since there isn't any expected interaction between the two options. Testing the last case every time would almost double the length of the test since the alluxio download is ~200mb and it gets downloaded both for the master and the worker.